crates/rover-client/src/operations/subgraph/fetch_all/runner.rs

```diff
@@ -0,0 +1,136 @@
+use graphql_client::*;
+
+use crate::blocking::StudioClient;
+use crate::operations::config::is_federated::{self, IsFederatedInput};
+use crate::RoverClientError;
+
+use super::types::*;
+
+#[derive(GraphQLQuery)]
+// The paths are relative to the directory where your `Cargo.toml` is located.
+// Both json and the GraphQL schema language are supported as sources for the schema
+#[graphql(
+    query_path = "src/operations/subgraph/fetch_all/fetch_all_query.graphql",
+    schema_path = ".schema/schema.graphql",
+    response_derives = "Eq, PartialEq, Debug, Serialize, Deserialize",
+    deprecated = "warn"
+)]
+/// This struct is used to generate the module containing `Variables` and
+/// `ResponseData` structs.
+/// Snake case of this name is the mod name. i.e. subgraph_fetch_all_query
+pub(crate) struct SubgraphFetchAllQuery;
+
+/// For a given graph return all of its subgraphs as a list
+pub fn run(
+    input: SubgraphFetchAllInput,
+    client: &StudioClient,
+) -> Result<Vec<Subgraph>, RoverClientError> {
+    // This response is used to check whether the current graph is federated.
+    let is_federated = is_federated::run(
+        IsFederatedInput {
+            graph_ref: input.graph_ref.clone(),
+        },
+        client,
+    )?;
+    if !is_federated {
+        return Err(RoverClientError::ExpectedFederatedGraph {
+            graph_ref: input.graph_ref,
+            can_operation_convert: false,
+        });
+    }
+    let variables = input.clone().into();
+    let response_data = client.post::<SubgraphFetchAllQuery>(variables)?;
+    get_subgraphs_from_response_data(input, response_data)
+}
+
+fn get_subgraphs_from_response_data(
+    input: SubgraphFetchAllInput,
+    response_data: SubgraphFetchAllResponseData,
+) -> Result<Vec<Subgraph>, RoverClientError> {
+    if let Some(maybe_variant) = response_data.variant {
+        match maybe_variant {
+            SubgraphFetchAllGraphVariant::GraphVariant(variant) => {
+                if let Some(subgraphs) = variant.subgraphs {
+                    Ok(subgraphs
```

Comment:
not a blocker, but wondering if this could be more readable with a `.and_then(|variant| variant.map(/*blah*/))`; don't waste time on it, though, because we can always just make it more readable later (mostly worried about those chunky `if let ... else` statements, which tend to benefit from `.and_then()`

but yeah, not a priority

Reply:
Had a crack at refactoring this, I've collapsed the if/let and the inner match statement together, because there's no point having both when you could just have a little more complexity in the pattern, and then gone with a `map_or_else`. Let me know if you think it's better or we need a second pass at it :)